### PR TITLE
Cleanup and refactor

### PR DESCRIPTION
**Pull request by ZeldaZach**
Signed-off-by: Zach Halpern <ZaHalpern+github@gmail.com>

## Related Ticket(s)
- Fixes #IssueNumber

## What will change with this Pull Request?
- this
- and this

### CODE REVIEW

**Comment by ebbit1q** [comment]
Terrible pull request description.

File: cockatrice/src/cardinfotext.cpp [hunk 1]
```diff
@@ -48,16 +48,16 @@ void CardInfoText::setCard(CardInfoPtr card)
 
     auto relatedCards = card->getRelatedCards();
     auto reverserelatedCards2Me = card->getReverseRelatedCards2Me();
-    if (relatedCards.size() || reverserelatedCards2Me.size()) {
+    if (!relatedCards.empty() || !reverserelatedCards2Me.empty()) {
         text += QString("<tr><td>%1</td><td width=\"5\"></td><td>").arg(tr("Related cards:"));
 
-        for (int i = 0; i < relatedCards.size(); ++i) {
-            QString tmp = relatedCards.at(i)->getName().toHtmlEscaped();
+        for (auto relatedCard : relatedCards) {
```
Comment (ebbit1q) [inline cockatrice/src/cardinfotext.cpp]: auto or const auto &?

Comment (ebbit1q) [reply cockatrice/src/cardinfotext.cpp]: actually it's a pointer nvm

File: cockatrice/src/filter_string.h [hunk 2]
```diff
@@ -26,7 +27,7 @@ class FilterString
     explicit FilterString(const QString &exp);
     bool check(CardData card)
     {
-        return result(card);
+        return result(std::move(card));
```
Comment (ebbit1q) [inline cockatrice/src/filter_string.h]: instead of std::move this should use a CardData reference

Comment (basicer) [reply cockatrice/src/filter_string.h]: CardData is a smart pointer anyhow, so Im not sure your saving anything by moving or referencing it.

**Pull request merged**